Use native time.Time types in firestore and rundex

wbxyz opened the pull request:
We were inconsistent with timezone and Unix vs UnixMilli previously, causing a lot of bad data in our metadata storage.

After doing some data migrations, firestore contains the data as native time.Time types now.

I also removed the rundex.Rebuild.Created field, because that's now directly handled in embed schema.Attempt

msuozzo commented:
I think we should revisit using a structured type here. I believe time.Time has the support we need from firestore and it certainly presents the best programming interface for us. Does that sound feasible? We'll likely need some migration code but it might be worthwhile in the long run.

wbxyz commented:
Added migration capabilities in #490 and executed them to clean up most of our data. I changed this PR to use native types as well, once this is submitted and deployed I'll need to migrate any entries created in the interval inbetween, then I'll re-enable the index on "created" fields.

msuozzo commented on pkg/rebuild/schema/schema.go:
```diff
@@ -257,17 +257,17 @@ type RebuildAttempt struct {
+	BenchmarkName string    `firestore:"benchmark_name,omitempty"`
+	BenchmarkHash string    `firestore:"benchmark_hash,omitempty"`
+	Type          string    `firestore:"run_type,omitempty"`
+	Created       time.Time `firestore:"created,omitempty"` // The time this run was created
```
remove comment
